subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -460,5 +472,37 @@ public FileCollection getAnnotationProcessorPath() {
     public void setAnnotationProcessorPath(@Nullable FileCollection annotationProcessorPath) {
         this.annotationProcessorPath = annotationProcessorPath;
     }
-}
 
+    /**
+     * Returns the directory to place source files generated by annotation compilers.
+     *
+     * @since 4.3
+     */
+    @Optional
+    @Nullable
+    @Incubating
+    @OutputDirectory
+    public File getAnnotationProcessorGeneratedSourcesDirectory() {
```

Comment:
When using a `DirectoryVar`, we could return it here. No need to get the `File` from the `DirectoryVar`/`PropertyState`.

Reply:
I guess we want to add some `PathSensitivity` annotation here, right?

Reply:
Not used for output properties.